x-pack/plugins/cloud_security_posture/public/pages/rules/rules_table.tsx

```diff
@@ -37,55 +50,166 @@ export const RulesTable = ({
   loading,
   error,
   selectedRuleId,
+  refetchStatus,
+  selectedRules,
+  setSelectedRules,
 }: RulesTableProps) => {
   const { euiTheme } = useEuiTheme();
-  const columns = useMemo(() => getColumns({ setSelectedRuleId }), [setSelectedRuleId]);
-
-  const euiPagination: EuiBasicTableProps<CspBenchmarkRule>['pagination'] = {
+  const euiPagination: EuiBasicTableProps<CspBenchmarkRulesWithStatus>['pagination'] = {
     pageIndex: page,
     pageSize,
     totalItemCount: total,
     pageSizeOptions: [10, 25, 100],
   };
 
-  const onTableChange = ({ page: pagination }: Criteria<CspBenchmarkRule>) => {
+  const onTableChange = ({ page: pagination }: Criteria<CspBenchmarkRulesWithStatus>) => {
     if (!pagination) return;
     setPagination({ page: pagination.index, perPage: pagination.size });
   };
 
-  const rowProps = (row: CspBenchmarkRule) => ({
-    ['data-test-subj']: TEST_SUBJECTS.getCspBenchmarkRuleTableRowItemTestId(row.metadata.id),
+  const rowProps = (row: CspBenchmarkRulesWithStatus) => ({
+    ['data-test-subj']: TEST_SUBJECTS.getCspBenchmarkRuleTableRowItemTestId(row.metadata?.id),
     style: {
-      background: row.metadata.id === selectedRuleId ? euiTheme.colors.highlight : undefined,
+      background: row.metadata?.id === selectedRuleId ? euiTheme.colors.highlight : undefined,
     },
   });
 
+  const [selectAllRulesThisPage, setSelectAllRulesThisPage] = useState<boolean>(false);
+  const postRequestChangeRulesStatus = useChangeCspRuleStatus();
+
+  const isArraySubset = (
+    smallArr: CspBenchmarkRulesWithStatus[],
+    bigArr: CspBenchmarkRulesWithStatus[]
+  ) => {
+    let i: number = 0;
+    const newSmallArr = smallArr.map((e) => e.metadata);
+    const newBigArr = bigArr.map((e) => e.metadata);
+
+    while (i < newSmallArr.length) {
+      if (!newBigArr.includes(newSmallArr[i])) return false;
+      i++;
+    }
+    return true;
+  };
```

Comment:
can you explain what we trying to do here? perhaps i could suggest a simpler solution

Reply:
this function is basically just to check if an array is a subset of a bigger array, this is currently being used to determine whether we should check/uncheck the top checkbox (the one that selects all rules on current page) when users checks all of the rules in that page or not